### DISCUSSION TITLE
WinUI3 app is not launching

### DISCUSSION THREAD

**Discussion by pbasithmpy** [microsoft/WindowsAppSDK#3008] [Q&A]
I am using unpackaged app with winUI 3 framework. And it was working fine. but now I am getting an error saying like below when app is launched. I have installed windowsappruntime installer already. could anyone can help me fixing this

Exception shown in Event viewer:

![launch_exception](https://user-images.githubusercontent.com/75209676/193005325-dadb7fa3-ae2b-4b0f-8ed6-2a7ccb96c9e2.png)

**Comment by DarranRowe** [comment]
There is a known issue #2918 with later updates for Windows 10 are having issues with permissions on the files in the Windows App SDK package.

**Reply by pbasithmpy** [reply]
But it was working fine in my system. And without any windows updates all of a sudden it is causing launch issue. But for x86 build it is working

**Reply by DarranRowe** [reply]
I tried to phrase it in such a way to not imply that it is the act of installing any Windows updates that causes it. Just that Windows being updated to at least a certain version can trigger it. Obviously, that didn't work.

The issue seems to be related to the Windows App SDK runtime being installed on Windows 10 patched to at least July or August, and the ACLs getting messed up with the Your Phone app ACLs. This means that it could be the combination of a newer patch of Windows 10 and the installation of Your Phone that is causing it. The latest version of Your Phone does include the Windows App SDK self contained and since we don't know how Windows processes this kind of thing then it could be doing weird things via hard links to reduce disk space usage.

Also, it isn't like Your Phone is going to be using the latest Windows App SDK, so things could be working well. Then an update to Your Phone is released that updates the self contained Windows App SDK to the same version as the runtime that is installed on your system, and that breaks things. So, things working fine and then suddenly breaking doesn't rule out this being a Windows issue.

The above reasoning also gives a good explanation as to why only the x64 version is broken. On x64 versions of Windows, only the x64 version of Your Phone is installed. I have also noticed in Visual Studio that sometimes it reports that the Windows App SDK libraries are being loaded from the Your Phone application directory, even on Windows 11, so it is certainly related.

But one thing you can try to test this out is:
1) Remove the Your Phone application for all users.
2) Remove all of the Windows App SDK components for all users.
3) Install the Windows App SDK redistributable.
4) Test your application again.

At this point, since only the Windows App SDK should be installed then the ACLs shouldn't be messed up. But you could also verify how badly the ACLs are messed up if you wanted to.

**Reply by pbasithmpy** [reply]
could you please help me on first two steps you suggested,
1. how to remove your phone application, I am not able to uninstall that
2. how to remove all windows app sdk components completely

**Reply by DarranRowe** [reply]
First, open up a powershell prompt with administrative rights.

For the Your Phone app.
Run:
`Get-AppxPackage -name *Phone*`

![Screenshot 2022-09-30 100418](https://user-images.githubusercontent.com/52577874/193235056-fcc8a8f0-ebb9-483d-b5d3-a8acd691d75e.png)

Copy the value that is listed on your system for the PackageFullName property. In the screenshot above it is Microsoft.YourPhone_1.22072.207.0_x64__8wekyb3d8bbwe, so I will be using that. But this might be different on your system.

Run:
`Remove-AppxPackage -AllUsers -Package Microsoft.YourPhone_1.22072.207.0_x64__8wekyb3d8bbwe`

I know Windows 11 has an issue where this may fail once. If it does fail, just try to run this command again. Also, if the package full name is different on your system, use that.

This should remove the Your Phone application for all users. This also gets rid of any caches too.

Next, for the Windows App SDK runtime.
Run:
`Get-AppxPackage -name *WinAppRuntime*`

For everything in this list remove them using Remove-AppxPackage -AllUsers.

Next, run:
`Get-AppxPackage -name *WindowsAppRuntime*`

For everything in this list remove them using Remove-AppxPackage -AllUsers.

If you did this correctly then everything should have been removed. The order that you remove the Windows App SDK runtimes is important. The DDLM, Singleton and Main packages all depend on the Windows App Runtime package. This means that the Windows App Runtime package has to be removed last.

**Reply by pbasithmpy** [reply]
I tried to remove your phone app. but it is failing to remove for all users. When I tried without all users, I still can not open the ui application. getting same Access denied error.
Error I am getting when I tried to remove your phone for all users is as below,


![your-phone-remove-error](https://user-images.githubusercontent.com/75209676/193988266-8bdc5a0b-5677-4913-aa4c-5b743c2f2d8f.png)

**Reply by pbasithmpy** [reply]
And also will microsoft fix this issue in any coming windows app sdk versions?

**Reply by DarranRowe** [reply]
Well, again I'll point out that this isn't a Windows App SDK problem. It is a Windows 10 problem.

![Screenshot 2022-10-05 142158](https://user-images.githubusercontent.com/52577874/194071252-87a322f1-9288-4263-aa9d-6be3b3ff39ab.png)

![Screenshot 2022-10-05 142218](https://user-images.githubusercontent.com/52577874/194071313-d3c0c4d8-8e6e-4151-b9d4-d01232dfba5a.png)

Windows 11 has no issues, and this is with the Phone Link (YourPhone) app installed.
The problem stemmed from the fact that when Windows deploys a packaged application, Windows will attempt to save space. It is possible to hard link files together, meaning there is only one instance of that file physically on disk. This is what Windows does for files in the SxS directory too.
Now, it seems as if Windows 10 was hard linking the Windows App Runtime files to the YourPhone files, which contained the Windows App SDK in a self contained form. But when it did this, Windows 10 messed up the access control lists for the files.

Any new release of the Windows App SDK may end up working, but if the issue in Windows 10 isn't fixed then any apps which contain the Windows App SDK in a self contained form could result in the same issue occurring. So, the most important thing is Windows 10 being fixed.

For your post about Remove-AppxPackage failing. The important thing is to reset the ACL for the Windows App SDK redistributables, so removing the Phone app is required to get rid of the potential conflict with the files.
So right now, if you open an administrator powershell console and run `Get-AppxPackage -AllUsers -Name *YourPhone*`, what output do you get?

The important things to look out for is whether it is actually installed, what users it is installed for, and whether or not it is staged. Also, check if it is provisioned in the image using `Get-ProvisionedAppxPackage -Online`. If you have successfully removed the Phone application completely, then it is fine if it failed for all users.

Also, the whole reason why I originally stated that you needed to remove the Phone application and completely remove the Windows App Runtime was to reset the ACLs and make sure that Windows wouldn't break them again. That is why it was important to completely and successfully remove everything.

**Reply by pbasithmpy** [reply]
for Get-AppxPackage -AllUsers -Name *YourPhone* this command I am getting an output like this
![your_phone_app](https://user-images.githubusercontent.com/75209676/194224481-bca6eeb7-3b66-4080-bbcf-1e7b4e3f2b5e.png)

but I am unable to delete the package

**Reply by pbasithmpy** [reply]
And I tried to built my application in unpackaged self contained mode, but that is also getting some sort of error when I publish the application. WinUI3 unpackaged self contained is not supported?
on publishing I am getting an error saying,
Microsoft.WindowsAppSDK.SelfContained.targets(42,5): error : The platform 'AnyCPU' is not supported for Self Contained mode.
even though I tried to publish the app for x64 platform

**Reply by DarranRowe** [reply]
Well, trying this out on my system was a pain. The store wasn't offering that version of the phone application to me by default. I had to do... things... in order to get the same version.

Anyway:

![Screenshot 2022-10-06 104641](https://user-images.githubusercontent.com/52577874/194282984-41d70ca2-bc25-415a-818d-6682a6a5428e.png)

there were no issues removing it. The fact that it is failing with ERROR_FILE_NOT_FOUND for you is also curious. I would honestly consider whether the package itself is corrupted and try repairing it.

Also, unpackaged self contained is supported. But I honestly can't say too much about the .NET project system because I use C++.

**Reply by pbasithmpy** [reply]
Thanks @DarranRowe for help. this was exactly the problem. I couldn't solve that by uninstalling your phone. because it is not always possible to make users to uninstall any application in their machine. So I changed my win ui app to windows app sdk self contained mode. And now there is no clash with any other apps and working perfectly

**Comment by gautambjain** [comment]
I received the same error all of a sudden. Then I discovered that x86 build does work. But x64 does not. Since then I have been only building x86 version.

**Reply by pbasithmpy** [reply]
yes. I tried x86 and it is working fine for me. But in my application there is dependency to x64 libraries. so I would need x64 build
